### PR TITLE
Docs hardwarefirewall

### LINKED ISSUE #2605: pfSense SG-2240 has reached End of Sale; consider alternatives like the SG-3100
# Documentation update request

## Description

Netgate, makers of the [SG-2240](https://store.netgate.com/SG-2440.aspx) we [recommend in the SecureDrop documentation](https://docs.securedrop.org/en/stable/network_firewall.html), has announced the end of sale of their SG-2240. They recommend the [SG-3100](https://store.netgate.com/SG-3100.aspx) as an alternative.

From a glance at their product page, it appears to be cheaper and perfectly capable for a SecureDrop instance. Of course, testing will have to be done on hardware before it can be officially recommended.

If the SG-3100 isn't a viable alternative, then other options will have to be considered.

## User Stories

As an administrator of a future SecureDrop installation, I would feel comfortable if the documentation recommended hardware that was still available for purchase, if I didn't prefer to use my own firewall of choice instead.

### PR DESCRIPTION
**Pull request by b-meson**
## Status

Ready for review.

## Description of Changes

Fixes #2605

### If you made changes to documentation:

- [x] Doc linting passed locally

### Notes

The `HTTP Referer` section that I added requires testing against pfSense hardware, preferably the SG-4860. I also ran into the same issue with the `SG-3100`. Therefore, I propose whoever reviews this also checks that they run into this issue. 

Additionally, I have updated the `.xml` files to the latest pfSense exports.

### CODE REVIEW

File: docs/network_firewall.rst [hunk 1]
```diff
@@ -177,6 +177,19 @@ Connect to the pfSense WebGUI
 
 .. _intentionally disables LAN access: https://labs.riseup.net/code/issues/7976
 
+Disable Refer Check
+~~~~~~~~~~~~~~~~~~~
+
+Before you can set up the hardware firewall, you will need to disable the
+"Browser HTTP Referer" setting after you login. You will see the Setup Wizard
+but you shoud exit out of it by clicking the pfSense logo at the top left
+corner. Navigate to **System** -> **Advanced** and look for the *Browser HTTP_Referer*
```
Comment (singuliere) [inline docs/network_firewall.rst]: \*\*Browser HTTP_REFERER\*\*

File: docs/network_firewall.rst [hunk 2]
```diff
@@ -177,6 +177,19 @@ Connect to the pfSense WebGUI
 
 .. _intentionally disables LAN access: https://labs.riseup.net/code/issues/7976
 
+Disable Refer Check
+~~~~~~~~~~~~~~~~~~~
+
+Before you can set up the hardware firewall, you will need to disable the
+"Browser HTTP Referer" setting after you login. You will see the Setup Wizard
```
Comment (singuliere) [inline docs/network_firewall.rst]: \*\*Browser HTTP_REFERER enforcement\*\*

File: docs/network_firewall.rst [hunk 3]
```diff
@@ -177,6 +177,19 @@ Connect to the pfSense WebGUI
...
+but you shoud exit out of it by clicking the pfSense logo at the top left
+corner. Navigate to **System** -> **Advanced** and look for the *Browser HTTP_Referer*
+enforcement setting. You should check it and then save your settings. After
+doing so, you should be able to go back to System and select Setup Wizard.
```
Comment (singuliere) [inline docs/network_firewall.rst]: \*\*System\*\*
\*\*Setup Wizard\*\*

**Comment by b-meson** [comment]
@singuliere thank you for your feedback. I have pushed a commit and updated my PR on the latest `develop`  for style edits.

**Comment by redshiftzero** [comment]
Hey @msheiny - I vaguely recall that you have one of these SG-4860 firewalls - can you confirm the HTTP-Referer section?

File: docs/network_firewall.rst [hunk 4]
```diff
@@ -177,6 +177,20 @@ Connect to the pfSense WebGUI
 
 .. _intentionally disables LAN access: https://labs.riseup.net/code/issues/7976
 
+Disable Refer Check
```
Comment (redshiftzero) [inline docs/network_firewall.rst]: Nit: `Refer` -> `Referer`

File: docs/network_firewall.rst [hunk 5]
```diff
@@ -177,6 +177,20 @@ Connect to the pfSense WebGUI
...
+~~~~~~~~~~~~~~~~~~~
+
+Before you can set up the hardware firewall, you will need to disable the
+**Browser HTTP_REFERER enforcement** setting after you login. You will see
+the Setup Wizard but you shoud exit out of it by clicking the pfSense logo
+at the top left corner. Navigate to **System** -> **Advanced** and look for
+the **Browser HTTP_REFERER enforcement** enforcement setting. You should check
+it and then save your settings. After doing so, you should be able to go
+back to **System** and select **Setup Wizard**.
```
Comment (redshiftzero) [inline docs/network_firewall.rst]: Hmm, why does this need to be disabled? Did you attempt to use the Alternate hostnames field such that you don't need to disable this enforcement?

File: install_files/network_firewall/filter-config-pfSense.xml [hunk 6]
```diff
@@ -1 +1,459 @@
...
+		<statetimeout></statetimeout>
+		<statetype><![CDATA[keep state]]></statetype>
+		<os></os>
+		<source>
+			<network>opt2</network>
+		</source>
+		<destination>
+			<network>opt1</network>
+		</destination>
+		<descr><![CDATA[Block all non-whitelisted traffic from OPT2 and OPT2]]></descr>
```
Comment (redshiftzero) [inline install_files/network_firewall/filter-config-pfSense.xml]: I inspected each line in the rules and found a nit: `OPT2 and OPT2` -> `OPT2 and OPT1`, fixed in followup commit 91ffda1

**Pull request merged**